### PR TITLE
Get recipient from notification rather than UserManager

### PR DESCRIPTION
**Pull request by aquibbaig**

### CODE REVIEW

File: site/src/Consumer/EmailNotificationConsumer.php [hunk 1]
```diff
@@ -82,9 +80,9 @@ protected function shouldHandle()
      */
     public function handle(): bool
     {
-        echo "Email Notification Consumer";
+        echo "Email Notification Consumer\n";
```
Comment (imphil) [inline site/src/Consumer/EmailNotificationConsumer.php]: You can remove this echo statement.

File: site/src/Consumer/AbstractNotificationConsumer.php [hunk 2]
```diff
@@ -72,6 +81,11 @@ public function setNotification(Notification $notification)
         $this->notification = $notification;
     }
 
+    public function setRecipient(User $user)
+    {
+        $this->recipient = $user;
+    }
+
```
Comment (imphil) [inline site/src/Consumer/AbstractNotificationConsumer.php]: You have this information already available in `$notification`, no need for a separate `$recipient` variable. (Actually, it's harmful to have a second copy of the same information, as the two could get out of sync.)

File: site/src/Consumer/EmailNotificationConsumer.php [hunk 3]
```diff
@@ -59,16 +59,14 @@ public function __construct(
 
     protected function shouldHandle()
     {
-        $user = $this->userManager->findUserByUsername($this->notification->getRecipient()->getUsername());
-
-        $emailSubscription = $user->isSubscribedToEmailNotifications();
+        $emailSubscription = $this->recipient->isSubscribedToEmailNotifications();
```
Comment (imphil) [inline site/src/Consumer/EmailNotificationConsumer.php]: Use `$this->notification->getRecipient()` instead.

**Comment by aquibbaig** [comment]
Thanks @imphil for your review on this PR. I have updated the branch now as per your review.

File: site/src/Consumer/AbstractNotificationConsumer.php (hunk 2 above)
Comment (aquibbaig) [reply site/src/Consumer/AbstractNotificationConsumer.php]: Yep

**Comment by imphil** [comment]
Please have a closer look at the diff. There should be no changes in `AbstractNotificationConsumer.php`

**Comment by aquibbaig** [comment]
@imphil I had forgotten to `git reset HEAD` the abstractNotificationConsumer file. Should be fine now!

**Pull request merged**